### DISCUSSION TITLE
Supporting more than 100 development iPhones

### DISCUSSION THREAD

**Discussion by alvarhansen** [MobileNativeFoundation/discussions#159] [General]
I've run into interesting problem at my current company. We have filled our 100 device limit on Apple Developer program. Removing them is not an option as we actually need more than 100 devices across the company for development. 

As this community has members from companies that have more than 100 iOS developers, I would love to hear what kind of set up you have to support debugging apps from more than 100 devices?

**Comment by chris-wickens** [comment]
Interesting dilemma. When you say for development do you definitely mean needed for debugging on a physical device? Otherwise I'm thinking you might be able to get away with using TestFlight for e.g. manual testing, UAT and defaulting to simulators for development except for cases where a physical device is needed. I agree it's an arbitrary limitation though and would be curious to hear others' solutions.

**Reply by alvarhansen** [reply]
> When you say for development do you definitely mean needed for debugging on a physical device?

Yes, the ability to debug/deploy apps directly from Xcode to the device. So, the basic developer workflow.

> I'm thinking you might be able to get away with using TestFlight for e.g. manual testing

Yep, thats what we do at the moment for QA flow. After merge to trunk, we build new app, deploy it to TestFlight and QA can access it though that. 
(Tip for people who do not know that: If you keep your app version same and only bump build number, you don't need to go through review again.)

**Comment by keith** [comment]
You can ask Apple for additional devices for this case, which I know many companies have done successfully

**Reply by alvarhansen** [reply]
Thanks for the insight! I've asked Apple for it and they keep saying it is not possible and they have nothing more to comment. But it is good to know that it can be done, so I just need to do better convincing.

**Comment by pmanuelli** [comment]
We have the same problem in my company. This is very annoying and it's even more working remotely: You no longer can borrow a device from a coworker.

We used an Enterprise version of our apps for debugging and internal distribution. But Enterprise apps have some limitations that were a problem for us. For example, it doesn't have In-App Purchases.

For people that don't need a testing device (only need a device to run some non-released version from time to time), we use Testflight. It takes longer to have de build available to download but that way we can free some devices for development.

Also keep in mind that the limitation is per device type (100 iPhones, 100 iPads, 100 iPods). We started using more iPads for development than we used to. It's not ideal, but it's something.

I would love to see too what other companies do :)

**Comment by d2callpod** [comment]
I haven't been involved in iOS for a long time so I don't know if this is relevant. [Test Fairy](testfairy.com) might be worth trying. We use it for Android and it works well. They do support iOS.
